### PR TITLE
Fixed "download and verify checksum" part of release CI

### PR DESCRIPTION
**Pull request by SergeyKleyman**
- Removed references to RELEASE_ID since it's not used (TAG_NAME is used instead)
- Added missing part of verifying downloaded binaries checksums

### CODE REVIEW

File: .ci/download_release_and_verify_checksum.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+this_script_dir="$( dirname "${BASH_SOURCE[0]}" )"
+this_script_dir="$( realpath "${this_script_dir}" )"
+
+ls -l "${original_packages_location}"
+
+echo "Downloading artifacts for tag \'${release_tag}\' to \'${downloaded_packages_location}\' ..."
+mkdir -p "${downloaded_packages_location}"
+pushd "${downloaded_packages_location}"
+
+"${this_script_dir}/run_command_with_timeout_and_retries.sh" --timeout=10 --max-tries=3 --wait-time-before-retry=10 --retry-on-error=yes -- gh release download "${release_tag}"
```
Comment (v1v) [inline .ci/download_release_and_verify_checksum.sh:17]:
```suggestion
if [ -f /usr/local/bin/bash_standard_lib.sh ] ; then
  source /usr/local/bin/bash_standard_lib.sh
  retry 3 gh release download "${release_tag}"
else
  gh release download "${release_tag}"
fi
```

File: .ci/run_command_with_timeout_and_retries.sh [hunk 2]
```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env bash
```
Comment (v1v) [inline .ci/run_command_with_timeout_and_retries.sh:1]: This particular script could be replaced by some functions which are available in the CI workers:

- https://github.com/elastic/apm-agent-php/blob/325bbac7d4d46e30de69e4509ab559e487509790/.ci/packer_cache.sh#L4
- https://github.com/elastic/apm-agent-php/blob/325bbac7d4d46e30de69e4509ab559e487509790/.ci/packer_cache.sh#L20

If so, see my comment above

**Review by v1v — COMMENTED** [review]
A minor comment regarding the `retry` and whether we could use the retry provided by the CI team for the shake of simplicity.

**Review by v1v — APPROVED** [review]

File: .ci/download_release_and_verify_checksum.sh (hunk 1 above)
Comment (SergeyKleyman) [reply .ci/download_release_and_verify_checksum.sh:17]: As we discussed on Slack the retry provided by the CI team doesn't have all the required functionality. In our case we need retry implementation that can distinguish between timeout and error - in some cases we need to retry on both timeout and error and in some cases we need to retry only on timeout.

Comment (SergeyKleyman) [reply .ci/download_release_and_verify_checksum.sh:17]: Another issue with using these shared libraries is complication with using it locally - we will need to fetch those script from infra GitHub repo to cover use case when host machine is not a CI worker. Is there better way to address this concern (in addition to the one in the comment above)?

File: .ci/run_command_with_timeout_and_retries.sh (hunk 2 above)
Comment (SergeyKleyman) [reply .ci/run_command_with_timeout_and_retries.sh:1]: Thank you for drawing my attention to this library. But there are some concerns - please see my comments above.

**Pull request merged**